index.js

```diff
@@ -51,27 +44,38 @@ function uglifyify(file, opts) {
     )
 
     debug = opts.sourceMap !== false && (debug || matched)
-    opts  = extend({}, {
+
+    var thisopts  = extend({}, {
       sourceMap: {
         filename: file
       }
     }, opts)
 
-    if (typeof opts.compress === 'object') {
-      delete opts.compress._
+    // remove exts before passing opts to uglify
+    delete thisopts.global
+    delete thisopts.exts
+    delete thisopts.x
+    delete thisopts._
+    delete thisopts._flags
+
+    if (typeof thisopts.compress === 'object') {
+      delete thisopts.compress._
     }
 
-    if (debug) opts.sourceMap.url = 'out.js.map'
+    if (debug) thisopts.sourceMap.url = 'out.js.map'
 
     // Check if incoming source code already has source map comment.
     // If so, send it in to ujs.minify as the inSourceMap parameter
     if (debug && matched) {
-      opts.sourceMap.content = convert.fromJSON(
+      thisopts.sourceMap.content = convert.fromJSON(
         new Buffer(matched[1], 'base64').toString()
       ).sourcemap
     }
 
-    var min = ujs.minify(buffer, opts)
+    var min = ujs.minify(buffer, thisopts)
+
+    if (min.error)
```

Comment:
okay, I was aware of what `through()` was doing, didn't realise that it was generating a stream instance ... although looking at the code I think that due to the implementation of the `capture()` function that doing `throw min.error` actually results in an "error" event being emitted. 

It is also worth considering that `min.error` might not be an actual `Error` instance ... "best practice" states you should only be throwing actual instances of `Error` (even though you can throw anything you like) .. so we might wish to transform the given `min.error` into a real `Error` object (note that `min.error` may be set by uglyfy's dependencies and that it's type may change depending on what the error is and where it originated.)